src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -182,6 +182,11 @@ public Conversion ClassifyConversionFromExpression(BoundExpression sourceExpress
             Debug.Assert(sourceExpression != null);
             Debug.Assert((object)destination != null);
 
+            if (sourceExpression.Type?.SpecialType == SpecialType.System_Void || destination.SpecialType == SpecialType.System_Void)
+            {
+                return Conversion.NoConversion;
```

Comment:
>return Conversion.NoConversion; [](start = 16, length = 31)

Do we need to add similar protection for other entry points in this class. For example, ```ClassifyConversionFromType``` below, etc.?

Reply:
In principle I think yes. However I'm struggling to come up with a program that actually exercises a bad behavior around `void` conversion through a call to `ClassifyConversionFromType` or other public methods on Conversions.